app/src/androidTest/java/org/mozilla/tv/firefox/helpers/ext/ViewInteraction.kt

```diff
@@ -10,7 +10,7 @@ import android.support.test.espresso.assertion.ViewAssertions.matches
 import android.support.test.espresso.matcher.ViewMatchers.isEnabled
 import org.hamcrest.CoreMatchers.not
 
-fun ViewInteraction.click() = this.perform(ViewActions.click())!!
+fun ViewInteraction.click(): ViewInteraction = this.perform(ViewActions.click())
```

Comment:
Personal preference, but I prefer double bangs for converting from platform types.  That conversion is dangerous, and `!!` communicates that to the reader.

Reply:
I hadn't considered that perspective before. However, it does goes against a deliberate Kotlin language decision to not treat all platform types as nullable, given what an inconvenient effect this has on developers. Then again, I also disagree with that decision because developers always need to think about it if a valuable is nullable or not and now the system just isn't helping us.

The reason for this change was that I originally found `!!` to be unclear: "why is `!!` necessary? What is this method returning? Oh, it's nullable? Okay, I guess that makes sense. If we wanted a non-nullable `ViewInteraction` to be returned, wouldn't it be clearer just to say that?" I thought the explicit type would help clarify that.

Maybe it's better to be doubly explicit? With a return type and !!? Then again, the final `!!` pattern could be learned. The explicit returns types are always redundant (and never go out of date) but they prevent you from making later mistakes when your expression return type is not what you expect.

I'll do both in this PR but let me know if you disagree.

Reply:
I think we should give some more thought to this and see if we come to a better conclusion, but both seems a good immediate solution.  I like having explicit return types on most functions anyway.